habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py

```diff
@@ -187,6 +192,46 @@ def reset(self, episode: Episode):
                 sim.robot.base_rot = angle_to_obj + np.pi / 2
             else:
                 sim.robot.base_rot = angle_to_obj
+        elif self._place_init:
+
+            if isinstance(sim.robot, StretchRobot):
+                sim.robot.arm_motor_pos = np.array(
+                    # [0.0] * 4 + [0.775, 0.0, -1.57000005, 0.0, 0.0, -0.7125] # gripper down
+                    [0.0] * 4 + [0.775, 0.0, 0.0, 0.0, 0.0, -0.7125] # gripper straight out
+                )
+                sim.robot.arm_joint_pos = np.array(
+                    # [0.0] * 4 + [0.775, 0.0, -1.57000005, 0.0, 0.0, -0.7125]
+                    [0.0] * 4 + [0.775, 0.0, 0.0, 0.0, 0.0, -0.7125]
+                )
+
+            spawn_recs = [
+                sim.receptacles[r]
+                for r in sim.receptacles.keys()
+                if r in sim.valid_goal_rec_names]
+            
+            snap_pos = np.array(
+                [
+                    r.get_surface_center(sim)
+                    for r in spawn_recs
+                ]
+            )
+            start_pos, angle_to_obj, was_succ = get_robot_spawns(
+                target_positions=snap_pos,
+                # rotation_perturbation_noise=self._config.base_angle_noise,
+                rotation_perturbation_noise=0.0,
+                distance_threshold=self._config.spawn_max_dists_to_obj,
+                sim=sim,
+                num_spawn_attempts=self._config.num_spawn_attempts,
+                physics_stability_steps=self._config.physics_stability_steps,
+            )
+
+            if was_succ:
```

Comment:
Should it be `if not was_succ:`

Reply:
It seems like `get_robot_spawns()` is returning True for when the spawn wasn't successful – which is also odd.

The most minimal change I can make for now is to change `was_succ` to `was_unsucc`.

Reply:
Sure thanks